### DISCUSSION TITLE
How to get Retry-After or Request Rate Limit Headers

### DISCUSSION THREAD

**Discussion by bajce** [pnp/pnpcore#1471] [Q&A]
I want to read the Retry-After and rate limit headers so I can use them in my application. Is there a way to get these headers? Is there any listener I can append or something similar?

I think the current retry-after in the SDK is based on a single context. In most cases, the throttling depends on the user or application that reads the data and is tenant-specific.

**Comment by jansenbe** [comment]
@bajce : check https://pnp.github.io/pnpcore/using-the-sdk/basics-eventhub.html#using-the-eventhub. Also see https://github.com/pnp/pnpassessment/blob/d45c7566d93bf8e466f07608655280bcfe4037ec/src/PnP.Scanning/PnP.Scanning.Core/Services/ScanManager.cs#L40 for an implementation
